File: templates/javascript/content/handler/src/features.js
import { createTool, createResource, createPrompt } from '@fastertools/ftl-sdk';

// Define your tools
export const tools = [
  createTool({
    name: 'echo',
    description: 'Echo a message back to the user',
    inputSchema: {
      type: 'object',
      properties: {
        message: { 
          type: 'string', 
          description: 'Message to echo back' 
        }
      },
      required: ['message']
    },
    execute: async (args) => {
      const message = args.message || 'Hello, world!';
      return `Echo: ${message}`;
    }
  }),
  // Add more tools here
];

// Define your resources
export const resources = [
  // Example:
  // createResource({
  //   uri: 'file:///example.txt',
  //   name: 'Example File',
  //   description: 'An example text file',
  //   mimeType: 'text/plain',
  //   read: async () => {
  //     return 'File contents here';
  //   }
  // }),
];

// Define your prompts
export const prompts = [
  // Example:
  // createPrompt({
  //   name: 'greeting',
  //   description: 'Generate a greeting message',
  //   arguments: [
  //     { name: 'name', description: 'Name to greet', required: true }
  //   ],
  //   resolve: async (args) => {
  //     return [
  //       { role: 'user', content: `Please greet ${args.name}` },
  //       { role: 'assistant', content: `Hello, ${args.name}! How can I help you today?` }
  //     ];
  //   }
  // }),
];

File: templates/rust/content/handler/src/features.rs
use ftl_sdk::{create_tool, create_resource, create_prompt, json, Tool, Resource, Prompt, PromptMessage, PromptRole};

// Define your tools
pub fn tools() -> Vec<Tool> {
    vec![
        create_tool(
            "echo",
            "Echo a message back to the user",
            json!({
                "type": "object",
                "properties": {
                    "message": { 
                        "type": "string", 
                        "description": "Message to echo back" 
                    }
                },
                "required": ["message"]
            }),
            |args| {
                let message = args["message"].as_str().unwrap_or("Hello, world!");
                Ok(format!("Echo: {}", message))
            }
        ),
        // Add more tools here
    ]
}

// Define your resources
pub fn resources() -> Vec<Resource> {
    vec![
        // Example:
        // create_resource(
        //     "file:///example.txt",
        //     "Example File",
        //     || {
        //         Ok("File contents here".to_string())
        //     }
        // )
        // .with_description("An example text file")
        // .with_mime_type("text/plain"),
    ]
}

// Define your prompts
pub fn prompts() -> Vec<Prompt> {
    vec![
        // Example:
        // create_prompt(
        //     "greeting",
        //     |args| {
        //         let name = args["name"].as_str().unwrap_or("User");
        //         Ok(vec![
        //             PromptMessage {
        //                 role: PromptRole::User,
        //                 content: format!("Please greet {}", name),
        //             },
        //             PromptMessage {
        //                 role: PromptRole::Assistant,
        //                 content: format!("Hello, {}! How can I help you today?", name),
        //             },
        //         ])
        //     }
        // )
        // .with_description("Generate a greeting message")
        // .with_argument("name", "Name to greet", true),
    ]
}